kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/rest/OrganizationsApiImpl.java

```diff
@@ -101,6 +116,88 @@ public Response listServiceClasses(String organizationIdParam) {
+    OrganizationId organizationId = new OrganizationId(KuntaApiConsts.IDENTIFIER_NAME, organizationIdParam);
+    EventId eventId = new EventId(KuntaApiConsts.IDENTIFIER_NAME, eventIdParam);
+    
+    for (EventProvider eventProvider : getEventProviders()) {
+      Event event = eventProvider.findOrganizationEvent(organizationId, eventId);
+      if (event != null) {
+        return Response.ok(event)
+          .build();
+      }
+    }
+    
+    return Response.status(Status.NOT_FOUND)
+      .build();
+  }
+
+  @Override
+  public Response findOrganizationEventImage(String organizationIdParam, String eventIdParam, String imageIdParam) {
+    OrganizationId organizationId = new OrganizationId(KuntaApiConsts.IDENTIFIER_NAME, organizationIdParam);
+    EventId eventId = new EventId(KuntaApiConsts.IDENTIFIER_NAME, eventIdParam);
+    AttachmentId attachmentId = new AttachmentId(KuntaApiConsts.IDENTIFIER_NAME, imageIdParam);
+    
+    for (EventProvider eventProvider : getEventProviders()) {
+      Attachment attachment = eventProvider.findEventImage(organizationId, eventId, attachmentId);
+      if (attachment != null) {
+        return Response.ok(attachment)
+          .build();
+      }
+    }
+    
+    return Response.status(Status.NOT_FOUND)
+      .build();
+  }
+
+  @Override
+  public Response getOrganizationEventImageData(String organizationIdParam, String eventIdParam, String imageIdParam, Integer size) {
+    OrganizationId organizationId = new OrganizationId(KuntaApiConsts.IDENTIFIER_NAME, organizationIdParam);
+    EventId eventId = new EventId(KuntaApiConsts.IDENTIFIER_NAME, eventIdParam);
+    AttachmentId attachmentId = new AttachmentId(KuntaApiConsts.IDENTIFIER_NAME, imageIdParam);
+    
+    for (EventProvider eventProvider : getEventProviders()) {
+      AttachmentData attachmentData = eventProvider.getEventImageData(organizationId, eventId, attachmentId);
+      if (attachmentData != null) {
+        return Response.ok(attachmentData.getData(), attachmentData.getType())
+          .build();
+      }
+    }
+    
+    return Response.status(Status.NOT_FOUND)
+      .build();
+  }
+
+  @Override
+  public Response listOrganizationEventImages(String organizationIdParam, String eventIdParam) {
+    List<Attachment> result = new ArrayList<>();
+    OrganizationId organizationId = new OrganizationId(KuntaApiConsts.IDENTIFIER_NAME, organizationIdParam);
+    EventId eventId = new EventId(KuntaApiConsts.IDENTIFIER_NAME, eventIdParam);
+    
+    for (EventProvider eventProvider : getEventProviders()) {
+      result.addAll(eventProvider.listEventImages(organizationId, eventId));
+    }
+    
+    return Response.ok(result)
+      .build();
+  }
+
+  @Override
+  public Response listOrganizationEvents(String organizationIdParam) {
+    OrganizationId organizationId = new OrganizationId(KuntaApiConsts.IDENTIFIER_NAME, organizationIdParam);
+    
+    List<Event> result = new ArrayList<>();
+   
+    // TODO: Sort events 
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)